plugins/outputs/cloudwatchlogs/pusher.go

```diff
@@ -82,6 +90,32 @@ func NewPusher(target Target, service CloudWatchLogsService, flushTimeout time.D
 	return p
 }
 
+func createLogger(p *pusher) {
+    stdout := zapcore.AddSync(os.Stdout)
+
+    level := zap.NewAtomicLevelAt(zap.InfoLevel)
+
+    productionCfg := zap.NewProductionEncoderConfig()
+    productionCfg.TimeKey = "timestamp"
+    productionCfg.EncodeTime = zapcore.ISO8601TimeEncoder
+    productionCfg.StacktraceKey = "stack"
+
+    jsonEncoder := zapcore.NewJSONEncoder(productionCfg)
+
+    jsonOutCore := zapcore.NewCore(jsonEncoder, stdout, level)
+
+    // Create a logger that samples every Nth message after the first M messages every S seconds
+    // where N = sc.Thereafter, M = sc.Initial, S = sc.Tick.
+    samplingCore := zapcore.NewSamplerWithOptions(
+        jsonOutCore,
+        time.Second, // interval
+        3, // log first 3 entries
+        0, // thereafter log zero entires within the interval
+    )
+    p.SampledLog = zap.New(samplingCore)
+    return
+}
```

Comment:
A couple things. 
1. I think that the code for creating the Zap logger here should not be tied to the pusher code. You should pull it out to `/logger`. 
2. It's bad code smell to have this create the logger, configure it and also set the pusher's logger to it. If we're pulling the creation code out of this file, I think realistically all you need to do is add the logger to the constructor function for the pusher. 
3. This is creating a separate logger for each pusher. Is that acceptable?

Reply:
I don't think we should create a new logger just for one log line. Even if we wanted to use the `zap.Logger`, we should be passing it into the pusher and reusing the one created in the agent.

Reply:
I removed the Zap sampler since we are using the Telegraf.Logger for cloudwatch logs.